contracts/ExoticPositionalMarkets/ExoticPositionalMarketManager.sol

```diff
@@ -0,0 +1,739 @@
+        }
+        IExoticPositionalMarket(_marketAddress).cancelMarket();
+        resolverAddress[msg.sender] = safeBoxAddress;
+        if (cancelledByCreator[_marketAddress]) {
+            IExoticPositionalMarket(_marketAddress).claimWinningTicketOnBehalf(creatorAddress[_marketAddress]);
+        }
+        removeActiveMarket(_marketAddress);
+        emit MarketCanceled(_marketAddress);
+    }
+
+    function resetMarket(address _marketAddress) external onlyOracleCouncilAndOwner {
+        require(isActiveMarket(_marketAddress), "Market not active");
+        if (IExoticPositionalMarket(_marketAddress).paused()) {
+            require(msg.sender == owner, "only pDAO");
+        }
+        IExoticPositionalMarket(_marketAddress).resetMarket();
+        emit MarketReset(_marketAddress);
+    }
+
+    function sendRewardToDisputor(
+        address _market,
+        address _disputorAddress,
+        uint _amount
+    ) external onlyOracleCouncilAndOwner {
+        IExoticRewards(exoticRewards).sendRewardToDisputoraddress(_market, _disputorAddress, _amount);
+        // emit RewardSentToDisputorForMarket(_market, _disputorAddress, _amount);
+    }
+
+    function issueBondsBackToCreatorAndResolver(address _marketAddress) external nonReentrant {
+        require(
+            IExoticPositionalMarket(_marketAddress).canUsersClaim() || cancelledByCreator[_marketAddress],
+            "Not claimable"
+        );
+        require(
+            IThalesBonds(thalesBonds).getCreatorBondForMarket(_marketAddress) > 0 ||
+                IThalesBonds(thalesBonds).getResolverBondForMarket(_marketAddress) > 0,
+            "Bonds already claimed"
+        );
+        IThalesBonds(thalesBonds).issueBondsBackToCreatorAndResolver(_marketAddress);
+    }
+
+    function disputeMarket(address _marketAddress, address _disputor) external onlyOracleCouncil {
+        IThalesBonds(thalesBonds).sendDisputorBondToMarket(
+            _marketAddress,
+            _disputor,
+            IExoticPositionalMarket(_marketAddress).disputePrice()
+        );
+        require(!IExoticPositionalMarket(_marketAddress).paused(), "Market paused");
+        if (!IExoticPositionalMarket(_marketAddress).disputed()) {
+            IExoticPositionalMarket(_marketAddress).openDispute();
+        }
+    }
+
+    function closeDispute(address _marketAddress) external onlyOracleCouncilAndOwner {
+        if (IExoticPositionalMarket(_marketAddress).paused()) {
+            require(msg.sender == owner, "Only pDAO");
+        }
+        require(IExoticPositionalMarket(_marketAddress).disputed(), "Market not disputed");
+        IExoticPositionalMarket(_marketAddress).closeDispute();
+    }
+
+    function getActiveMarketAddress(uint _index) external view returns (address) {
+        return activeMarkets[_index];
+    }
+
+    function getActiveMarketIndex(address _marketAddress) public view returns (uint) {
+        for (uint i = 0; i < numOfActiveMarkets; i++) {
```

Comment:
are you sure this will work when there are 1000 markets? Usually for loops have limits when they reach a certain number. Why is it needed?